### DISCUSSION TITLE
Form View Add a Number that is 1 up after last Numer in the Year

### DISCUSSION THREAD

**Discussion by Peter1980FFM** [Budibase/budibase#15023] [Help] [answered]
Hallo,
I want to generate a Default Value for a field 
Like 
If Datefomat YYYY is in the Collum Year, than add 1 to the higest numer in Collum BeschlussID for the YYYY else 1  how can  I make this happen?

And the Second Question is is ther an an posibility that i can fill out in this Row a Collum like BeschlussNummer that have the format Collum ID "/" Collum Year?

Thank you for Helping me I just tried du find a Solution in the Dokumentation but didn´t find the right way

**Comment by mikesealey** [comment, marked as answer]
Hey @Peter1980FFM 

You'll want to use a Formula column - formula columns have access to the values stored in other cells of the same row, so you can do a little bit of JavaScript in order to add a year to the date stored in a row.

Firstly, add a formula column to your table, and set it's response to "Date/Time". 

Set the Formula to 

```JS
let date = new Date($("Favourite_year"))

date.setFullYear(date.getFullYear() + 1)

return new Date(date)
```
Here's an app-export if you want a closer look
[15023-export-1732555904685.tar.gz](https://github.com/user-attachments/files/17906771/15023-export-1732555904685.tar.gz)
